music-journal/utils/auth.js

```diff
@@ -1,15 +1,13 @@
-import { fetcher } from './fetcher';
+import Cookies from 'js-cookie'
+import { getMe } from './fetcher'
 
-function deserialize(serializedJavascript) {
-    return eval('(' + serializedJavascript + ')');
-}
-export default async function auth(cookie) {
-    const firebaseToken = deserialize(cookie.replace('__session=','')).firebaseToken;
-    const { data: { user }, status } = await fetcher(`/api/me?token=${firebaseToken}`);
-
-    if(status !== 200) {
-        return false;
+export default async function auth() {
+    try {
+        const { firebaseToken } = JSON.parse(Cookies.get('__session'))

```

Comment:
I'm so glad this is working, during the `cookie trials` I was thrashing with js-cookie, react-cookie and others trying to do this

Reply:
Well I never would have figured it out at all! Was easy to tidy up when you had it all working